packages/wrangler/src/cfetch/internal.ts

```diff
@@ -26,20 +27,36 @@ export async function fetchInternal<ResponseType>(
   addAuthorizationHeader(headers, apiToken);
 
   const queryString = queryParams ? `?${queryParams.toString()}` : "";
+  const method = init.method ?? "GET";
   const response = await fetch(`${CF_API_BASE_URL}${resource}${queryString}`, {
-    method: "GET",
+    method,
     ...init,
     headers,
   });
   const jsonText = await response.text();
   try {
-    const json = JSON.parse(jsonText);
-    return json as ResponseType;
-  } catch (e) {
-    throw new Error(
-      `Failed to fetch ${resource} - ${response.status}: ${response.statusText}\nInvalid JSON response:\n${jsonText}`
-    );
+    return parseJSON(jsonText) as ResponseType;
+  } catch (err) {
```

Comment:
I don't think so, it's easier to just show a truncated view of the non-JSON body. This should be extremely rare anyway, since it's an API contract violation.